cordova-lib/src/cordova/util.js

```diff
@@ -185,6 +187,31 @@ function listPlatforms(project_dir) {
     });
 }
 
+function getInstalledPlatformsWithVersions(project_dir) {
+    var platforms_on_fs = listPlatforms(project_dir);
+
+    return Q.all(platforms_on_fs.map(function(p) {
+        return superspawn.maybeSpawn(path.join(project_dir, 'platforms', p, 'cordova', 'version'), [], { chmod: true })
+        .then(function(v) {
+            return {
+                platform: p,
+                version: v ? v : null
+            };
+        }, function(v) {
+            return {
+                platform: p,
+                version: 'broken'
+            };
+        });
+    })).then(function(platformVersions) {
+        var result = {};
+        for(var i = 0; i < platformVersions.length; i++) {
+            result[platformVersions[i].platform] = platformVersions[i].version;
+        }
+        return result;
+    });
+}
```

Comment:
The above method would be simplified if you just accumulated the `result` object as you go... As in... initialize `result` before the `Q.all()` call, then instead of, for example, `return { platform: p, version: v ? v : null };` you would just do `result[p] = v || null;`. In the final then, you would ignore the parameter and just return `result`.